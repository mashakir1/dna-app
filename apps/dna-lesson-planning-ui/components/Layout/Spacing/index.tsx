// Styling
import styles from "./spacing.module.scss";

interface SpacingProps {
  height: "xs" | "s" | "m" | "l" | "xl" | "xxl";
}

const propToCssClassMap = {
  xs: styles.spacingXS,
  s: styles.spacingS,
  m: styles.spacingM,
  l: styles.spacingL,
  xl: styles.spacingXL,
  xxl: styles.spacingXXL
}

const Spacing: (props: SpacingProps) => JSX.Element = ({ height }) => {
  return <div className={propToCssClassMap[height]} />;
};

export default Spacing;
