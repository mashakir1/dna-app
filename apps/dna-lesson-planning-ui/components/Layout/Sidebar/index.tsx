// * Imports
import useTranslation from "next-translate/useTranslation";

//* Components
import { NordAngliaLogo, SpeechBubbleIcon } from "components/Svg";
import Spacing from "components/Layout/Spacing";
import NavigationButtonList from "./NavigationButtonList";
import { IconLinkButton } from "components/Buttons";

//* Styles
import sidebarStyling from "./sidebar.module.scss";
import feedbackStyling from "styles/components/sidebar/feedback.module.scss";
import navigationButtonStyling from "styles/components/sidebar/navigationButton.module.scss";

const Sidebar: () => JSX.Element = () => {
  const { t } = useTranslation("common");
  return (
    <div className={sidebarStyling.sidebar}>
      <div className={sidebarStyling.sidebarContent}>
        <NordAngliaLogo />
        <Spacing height={"xxl"} />
        <NavigationButtonList />
        <IconLinkButton
          href="/feedback"
          icon={<SpeechBubbleIcon />}
          text={t("feedback")}
          classNames={`${navigationButtonStyling.sidebarButton} ${feedbackStyling.feedback} flex justify-center items-center`}
        />
      </div>
    </div>
  );
};

export default Sidebar;
