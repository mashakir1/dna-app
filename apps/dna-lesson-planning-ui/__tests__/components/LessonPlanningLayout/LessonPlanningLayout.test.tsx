import { render } from "@testing-library/react";

import Layout from "components/LessonPlanningLayout";

describe("Layout", () => {
  it("should render the lesson planning layout", () => {
    const { container } = render(
      <Layout>
        <></>
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });
});
