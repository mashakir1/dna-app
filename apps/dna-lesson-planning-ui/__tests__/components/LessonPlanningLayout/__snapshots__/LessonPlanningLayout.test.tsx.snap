// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the lesson planning layout 1`] = `
<div>
  <div
    class="page"
  >
    <div
      class="header flex justify-between items-center"
    >
      <nav
        class="breadcrumbs flex"
      >
        <p
          aria-label="choose-unit"
          class="small breadcrumb active "
        >
          choose-unit
        </p>
        <p
          aria-label="journey"
          class="small breadcrumb  "
        >
          journey
        </p>
        <p
          aria-label="plan-lesson"
          class="small breadcrumb  "
        >
          plan-lesson
        </p>
        <p
          aria-label="review"
          class="small breadcrumb  "
        >
          review
        </p>
      </nav>
      <span>
        cancel
      </span>
    </div>
    <div
      class="container flex column gap-l"
    />
  </div>
</div>
`;
